spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -193,47 +193,442 @@ For example, calling an `@Outgoing` annotated method from user code will not pos
+----
+@Outgoing("channel") 
+Message<O> method()
+----
+| Produces a infinite streams of `Message` associated with the channel `channel`. 
+|	This method is called for each _request_ made by the subscriber.
+
+|
+[source, java]
+----
+@Outgoing("channel")
+O method()
+----
+| Produces an infinite streams of _payload_ associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation.
+|	This method is called for each _request_ made by the subscriber.
+
+|
+[source, java]
+----
+@Outgoing("channel")
+CompletionStage<Message<O>> method()
+----
+| Produces a infinite streams of `Message` associated with the channel `channel`. The result is a `CompletionStage`. The method cannot be called until the `CompletionStage` returned previously is completed.
+|	This method is called for each _request_ made by the subscriber.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+CompletionStage<O> method()
+----
+| Produces a infinite streams of _payload_ associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation. The result is a `CompletionStage`. The method cannot be called until the `CompletionStage` returned previously is completed.
+|	This method is called for each _request_ made by the subscriber.
+
+|===
+
+
+===== Methods consuming data
+
+[cols="2a,1,1",options="header"]
+|===
+|Signature
+|Behavior
+|Invocation
+
+|
+[source,java]
+----
+@Incoming("channel")
+Subscriber<Message<I>> method()
+----
+| Returns a `Subscriber` that receives the `Message` objects transiting on the channel `channel`.
+| The method is called only once to retrieve the `Subscriber` object at assembly time.
+ This subscriber is connected to the matching channel.
+
+|
+[source,java]
+----
+@Incoming("channel")
+Subscriber<I> method()
+----
+| Returns a `Subscriber` that receives the _payload_ objects transiting on the channel `channel`.
+The payload is automatically extracted from the inflight messages using `Message.getPayload()`.
+| The method is called only once to retrieve the `Subscriber` object at assembly time.
+ This subscriber is connected to the matching channel.
+
+|
+[source,java]
+----
+@Incoming("channel") 
+SubscriberBuilder<Message<I>> method()
+----
+| Returns a `SubscriberBuilder` that receives the `Message` objects transiting on the channel `channel`.
+| The method is called only once to retrieve the `Subscriber` object at assembly time. 
+The created `subscriber` is connected to the matching channel.
+
+|
+[source,java]
+----
+@Incoming("channel")
+SubscriberBuilder<I> method()
+----
+| Returns a `SubscriberBuilder` that receives the `Message` objects transiting on the channel `channel`.
+The payload is automatically extracted from the inflight messages using `Message.getPayload()`.
+| The method is called only once to retrieve the `Subscriber` object at assembly time.
+The created `subscriber` is connected to the matching channel.
```

Comment:
..that is used to build a CompletionSubscriber that is subscribed to the matching channel.